lib/ohai/mixin/os.rb

```diff
@@ -27,6 +27,65 @@ module OS
       #
       # @return [String] the OS
       def collect_os
+        if target_mode?
+          collect_os_target
+        else
+          collect_os_local
+        end
+      end
+
+      # This should exactly preserve the semantics of collect_os_local below, which is authoritative
+      # for the API and must adhere to pre-existing ohai semantics and not follow inspec's notion of
+      # os/family/hierarchy.
+      #
+      # Right or wrong the ohai `os` variable has matched the ruby `host_os` definition for the past
+      # 10+ years, preceding inspec/train's definitions and this is the docuemnted correct API of
+      # these methods.  Mismatches between the ruby notion and the train version will be fixed as
+      # bugfixes in these methods and may not be considered semver violating even though they make
+      # break downstream consumers.  Please ensure that both methods produce the same results if
+      # you are on a platform which supports running ruby (train is considered authoritative for
+      # any "OS" which cannot run ruby -- server consoles, REST APIs, etc...)
+      #
+      # @api private
+      def collect_os_target
+        case
+        when transport_connection.os.aix?
+          "aix"
+        when transport_connection.os.darwin?
+          "darwin"
+        when transport_connection.os.linux?
+          "linux"
+        when transport_connection.os.family == "freebsd"
+          "freebsd"
+        when transport_connection.os.family == "openbsd"
+          "openbsd"
+        when transport_connection.os.family == "netbsd"
+          "netbsd"
+        when transport_connection.os.family == "dragonflybsd"
+          "dragonflybsd"
+        when transport_connection.os.solaris?
+          "solaris2"
+        when transport_connection.os.windows?
+          "windows"
+
+          #
+          # The purpose of the next two lines is that anything which runs Unix is presumed to be able to run ruby, and
+          # if it was not caught above, we MUST translate whatever train uses as the 'os' into the proper ruby host_os
+          # string.  If it is not unix and not caught above we assume it is something like a REST API which cannot run
+          # ruby.  If these assumptions are incorrect then it is a bug, which should be submitted to fix it, and the
+          # values should not be relied upon until that bug is fixed.  The train os is NEVER considered authoritataive
+          # for any target which can run ruby.
+          #
+
+        when transport_connection.os.unix?
+          raise "Ohai target mode not supported on this host"
```

Comment:
We probably want to raise with a better message here that gives users a next step to get in contact with us to tell us what they're trying to use.